scheduler/stack.go

```diff
@@ -286,7 +293,7 @@ func NewSystemStack(sysbatch bool, ctx Context) *SystemStack {
 	}
```

Comment:
This block is using the scheduler config from the state, not the one we've set earlier. It's not a bug right now because we don't have preemption in the node pool scheduler config, but we'll need to address this when we do anyways so if it's not a huge lift now we might want to do that too. (Or just move finding `enablePreemption` into the `NewBinPackIterator`.)

Reply:
This, along with the part mentioned in https://github.com/hashicorp/nomad/pull/17598#discussion_r1235768274, are the preemption bits that I was reluctant to touch right now. 

Despite reading the full scheduler configuration, only the preemption bits are used, so, as you mentioned, it doesn't affects us now.

Fixing this is a bit awkward because the job (and therefore the node pool) is only known once `SetJob` is called, and so the [`evict`](https://github.com/hashicorp/nomad/blob/main/scheduler/rank.go#L178) flag in the `BinPackIterator` iterator could be modified mid-scheduling, and I don't have the full grasp of what this could cause.

Maybe for now we can add a comment here and in `selectNextOption` noting that only preemption values should be used because the node pool may override the others?

Reply:
> Maybe for now we can add a comment here and in `selectNextOption` noting that only preemption values should be used because the node pool may override the others?

Yeah, that seems reasonable for keeping the risk of this change contained.

Reply:
Added some comments in https://github.com/hashicorp/nomad/pull/17598/commits/6b07aedcaa56233bcc4e41f01c169a9b1f770f00